### PR TITLE
[connector/spanmetricsconnector] Rename the `operation` attribute to `span.name`.

### PR DESCRIPTION
**Pull request by Cluas**
**Description:** <Describe what has changed.>

 Rename the `operation` attribute to `span.name`

**Link to tracking Issue:** <Issue number if applicable>
#18529
**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by kovrus** [comment]
See comment about changelog entry: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/18744#issuecomment-1434706203

**Comment by Cluas** [comment]
> 

🤝 need help to add `Skip Changelog` label.

**Comment by Cluas** [comment]
@albertteoh @kovrus  PTAL

File: connector/spanmetricsconnector/README.md [hunk 1]
```diff
@@ -19,31 +19,31 @@ Aggregates Request, Error and Duration (R.E.D) metrics from span data.
 **Request** counts are computed as the number of spans seen per unique set of dimensions, including Errors.
 For example, the following metric shows 142 calls:
 ```
-calls_total{http_method="GET",http_status_code="200",operation="/Address",service_name="shippingservice",span_kind="SPAN_KIND_SERVER",status_code="STATUS_CODE_UNSET"} 142
+calls_total{http_method="GET",http_status_code="200",span_name="/Address",service_name="shippingservice",span_kind="SPAN_KIND_SERVER",status_code="STATUS_CODE_UNSET"} 142
 ```
-Multiple metrics can be aggregated if, for instance, a user wishes to view call counts just on `service_name` and `operation`.
+Multiple metrics can be aggregated if, for instance, a user wishes to view call counts just on `service_name` and `span_name`.
```
Comment (kovrus) [inline connector/spanmetricsconnector/README.md:24]: It should probably be:

```suggestion
Multiple metrics can be aggregated if, for instance, a user wishes to view call counts just on `service.name` and `span.name`.
```

**Review by kovrus — COMMENTED** [review]
I think we should just drop examples in the Prometheus exposition format instead of modifying them. Maybe just drop them instead of modification. We can provide documentation on how to export OTel metrics to Prometheus later.

File: connector/spanmetricsconnector/README.md (hunk 1 above)
Comment (Cluas) [reply connector/spanmetricsconnector/README.md:24]: I completely agree. We should raise a separate PR to update the documentation.

Comment (Cluas) [reply connector/spanmetricsconnector/README.md:24]: This definitely needs to be modified in the end. We can temporarily ignore the issue of compliance here. @kovrus What do you think

Comment (kovrus) [reply connector/spanmetricsconnector/README.md:24]: Yep, let's follow up on it later.

**Review by kovrus — APPROVED** [review]
For reviewers: please ignore the remaining that is done in Prometheus exp. format examples. It will be removed and documentation will be updated later.

**Pull request merged**